### DISCUSSION TITLE
Can not compile a project

### DISCUSSION THREAD

**Discussion by vitw** [hecrj/iced#1072] [Q&A] [answered]
Compiling wgpu-types v0.7.0
error[E0658]: `match` is not allowed in a `const fn`
    --> /home/vp/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-types-0.7.0/src/lib.rs:1844:9
     |
1844 | /         match self {
1845 | |             Self::Uchar2 | Self::Char2 | Self::Uchar2Norm | Self::Char2Norm => 2,
1846 | |             Self::Uchar4
1847 | |             | Self::Char4
...    |
1870 | |             Self::Double4 => 32,
1871 | |         }
     | |_________^
     |
     = note: see issue #49146 <https://github.com/rust-lang/rust/issues/49146> for more information
     = help: add `#![feature(const_if_match)]` to the crate attributes to enable

**Comment by PossiblyAShrub** [comment, marked as answer]
Try updating your rust compiler, and then recompile. If you are using rustup then run `rustup update`.
